src/seer/automation/codegen/retry_unittest_coding_component.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from langfuse.decorators import observe
+from sentry_sdk.ai.monitoring import ai_track
+
+from integrations.codecov.codecov_client import CodecovClient
+from seer.automation.agent.agent import AgentConfig, LlmAgent, RunConfig
+from seer.automation.agent.client import AnthropicProvider, LlmClient
+from seer.automation.autofix.components.coding.models import PlanStepsPromptXml
+from seer.automation.autofix.components.coding.utils import (
+    task_to_file_change,
+    task_to_file_create,
+    task_to_file_delete,
+)
+from seer.automation.autofix.tools import BaseTools
+from seer.automation.codebase.repo_client import RepoClientType
+from seer.automation.codegen.codegen_context import CodegenContext
+from seer.automation.codegen.models import CodeUnitTestOutput, CodeUnitTestRequest
+from seer.automation.codegen.prompts import CodingUnitTestPrompts, RetryUnitTestPrompts
+from seer.automation.component import BaseComponent
+from seer.automation.models import FileChange
+from seer.automation.utils import escape_multi_xml, extract_text_inside_tags
+from seer.db import DbPrContextToUnitTestGenerationRunIdMapping
+from seer.dependency_injection import inject, injected
+
+logger = logging.getLogger(__name__)
+
+
+class RetryUnitTestCodingComponent(BaseComponent[CodeUnitTestRequest, CodeUnitTestOutput]):
+    context: CodegenContext
+
+    @observe(name="Retry unit tests")
+    @ai_track(description="Retry unit test generation")
+    @inject
+    def invoke(
+        self,
+        request: CodeUnitTestRequest,
+        previous_run_context: DbPrContextToUnitTestGenerationRunIdMapping,
+        llm_client: LlmClient = injected,
+    ) -> CodeUnitTestOutput | None:
+        with BaseTools(self.context, repo_client_type=RepoClientType.CODECOV_PR_REVIEW) as tools:
+            previous_run_memory = self.context.get_memory(
+                "UnitTestRunMemory", past_run_id=previous_run_context.run_id
```

Comment:
is this getting the right memory key?
```suggestion
                "unit_test_memory", past_run_id=previous_run_context.run_id
```